plugins/inputs/http/http.go

```diff
@@ -116,13 +136,69 @@ func (h *HTTP) Init() error {
 
 	h.client = client
 
+	if h.CookieAuthURL != "" {
+		// cookie auth defaults
+		renewalInterval := defaultCookieAuthRenewal
+		if h.CookieAuthRenewal != 0 {
+			renewalInterval = time.Duration(h.CookieAuthRenewal)
+		}
+		if h.CookieAuthMethod == "" {
+			h.CookieAuthMethod = defaultCookieAuthMethod
+		}
+		// add cookie jar to HTTP client
+		if h.client.Jar, err = cookiejar.New(nil); err != nil {
+			return err
+		}
+		// start auth ticker
+		if authErr := h.startCookieAuth(renewalInterval); authErr != nil {
+			return authErr
+		}
+	}
+
 	// Set default as [200]
 	if len(h.SuccessStatusCodes) == 0 {
 		h.SuccessStatusCodes = []int{200}
 	}
 	return nil
 }
 
+func (h *HTTP) startCookieAuth(interval time.Duration) error {
+	// continual auth ticker
+	ticker := time.NewTicker(interval)
+	go func() {
+		for range ticker.C {
+			_ = h.doCookieAuth()
+		}
+	}()
```

Comment:
How about
```suggestion
	ticker := time.AfterFunc(interval, func() { _ = h.doCookieAuth() })
```

Reply:
I was under the impression that `time.AfterFunc()`  only fires once?

Reply:
Oh I think you are right. Sorry for the noise!